client/components/app.jsx

```diff
@@ -35,20 +38,57 @@ class App extends Component {
     });
   }
 
+  async getCartItems() {
+    try {
+      const response = await fetch('/api/cart');
+      const cart = await response.json();
+      this.setState({
+        cart
+      });
+    } catch (error) {
+      console.error(error.message);
+    }
+  }
+
+  async addToCart(product) {
+    try {
+      const productAdded = {
+        productId: product.productId
+      };
+      const headers = new Headers();
+      headers.append('Content-Type', 'application/json');
+      const response = await fetch('/api/cart', {
+        method: 'POST',
+        body: JSON.stringify(productAdded),
+        headers
+      });
+      const json = await response.json();
+      this.setState(previous => {
+        const updatedCart = previous.cart;
+        updatedCart.push(json);
```

Comment:
Be careful not to mutate your state. Arrays are objects, which are reference types. A new variable is not a copy.